File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/settings/AzureSettingsConfigurableProvider.java
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

package com.microsoft.azure.toolkit.intellij.settings;

import com.intellij.openapi.options.Configurable;
import com.intellij.openapi.options.ConfigurableProvider;
import org.jetbrains.annotations.Nullable;

public class AzureSettingsConfigurableProvider extends ConfigurableProvider {

    @Override
    public boolean canCreateConfigurable() {
        return true;
    }

    @Nullable
    @Override
    public Configurable createConfigurable() {
        return new AzureSettingsConfigurable();
    }
}

File: PluginsAndFeatures/azure-toolkit-for-rider/src/main/kotlin/com/microsoft/azure/toolkit/intellij/settings/AzureSettingsConfigurableProvider.kt
/*
 * Copyright 2018-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the MIT license.
 */

package com.microsoft.azure.toolkit.intellij.settings

import com.intellij.openapi.options.ConfigurableProvider

class AzureSettingsConfigurableProvider : ConfigurableProvider() {
    override fun canCreateConfigurable() = true
    override fun createConfigurable() = AzureSettingsConfigurable()
}